SimpleBrowser/Browser.cs

```diff
@@ -1281,11 +1289,15 @@ private IHttpWebRequest PrepareRequestObject(Uri url, string method, string cont
 		private void Register(Browser browser)
 		{
 			_allWindows.Add(browser);
+            if(!_allContexts.Contains(_allWindows)){
+                _allContexts.Add(_allWindows);
+            }
 			if (browser.WindowHandle == null)
 			{
 				browser.WindowHandle = Guid.NewGuid().ToString().Substring(0, 8);
 			}
 		}
+        private static List<List<Browser>> _allContexts = new List<List<Browser>>();
 

```

Comment:
Well, the old paradigm was that SimpleBrowser had only one "name-space". Much like real browsers do. If you open Internet Explorer (window 1) and click a link with target="myName" (opens in window 2) and then you start IE again (window 3), clicking that same link in your new window, the link will open in the same window 2. There is one process-wide list of named windows.

Because we want to be able to run concurrent unit tests that shouldn't interfere with each other, we now create a new name-space when you start a new Browser. When you click a link with a target attribute there, it will open in the same name-space. So your change from Browser.ClearWindows() to myBrowser.ClearWindows() will clear only the current name-space. But others still rely on the static method for clearing all namespaces.